src/main/java/org/traccar/protocol/ZrProtocolEncoder.java

```diff
@@ -0,0 +1,78 @@
+package org.traccar.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.traccar.BaseProtocolEncoder;
+import org.traccar.Protocol;
+import org.traccar.helper.DataConverter;
+import org.traccar.model.Command;
+
+/**
+ * @author QingtaiJiang
+ * @date 2023/9/19 14:17
+ * @email qingtaij@163.com
+ */
+public class ZrProtocolEncoder extends BaseProtocolEncoder {
+
+    public ZrProtocolEncoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    @Override
+    protected Object encodeCommand(Command command) {
+        ByteBuf id = Unpooled.wrappedBuffer(DataConverter.parseHex(strHexPaddingLeft(getUniqueId(command.getDeviceId()), 20)));
+        switch (command.getType()) {
+            case Command.TYPE_SET_CONNECTION:
+                return registrationInformation(id, command);
+            case Command.TYPE_ALARM_SPEED:
```

Comment:
What is this?

Reply:
Command.TYPE_SET_CONNECTION is concatenate the command for updating the IP and port.   Command.TYPE_ALARM_SPEED will update latter.

Reply:
Update later? Is this PR not ready for review?

Reply:
Please wait a moment, I'm still making the necessary adjustments.

Reply:
ready

Reply:
It doesn't look like all my comments are addressed. Once you fix everything let me know.

Reply:
I think I have made the necessary changes. Please, could you review again. If there are any remaining irregularities, kindly point them out, and I would greatly appreciate it.

Reply:
Unfortunately I don't think you addressed all my comments.